app/controllers/suggestions_controller.rb

```diff
@@ -3,38 +3,51 @@ class SuggestionsController < ApplicationController
 
   def index
     @suggestion = Suggestion.new
-    # if !params[:track_name].empty?
-    #   @track_name = RSpotify::Track.search(params[:track_name])
-    # else
+    if params[:query]
+      @result = RSpotify::Track.search(params[:query])
+    end
       render :index
-    # end
   end
 
   def create
-    # @suggestion = Suggestion.create(song_params)
-    if params[:query]
-      @result = RSpotify::Track.search(params[:query])
+    track = RSpotify::Track.find(params[:track_id])
+    artists = ""
+    if track.artists.length != 1
+      track.artists.each do |x|
+        artists = artists + x.name + ", "
+      end
+      artists = artists[0..-3]
+    else
+      artists = track.artists[0].name
```

Comment:
@evo21  lines 13-21 can also just be this one line:

artists = track.artists.map(&:name).join(', ')

Although what you put is perfectly valid.  When you have a collection of objects and you want to extract a list of a particular attribute from that collection (in this example you want all the names from your collection of artists) you can use the map method on arrays.  Normally you pass map a block, and map will return an array of the result of the block.

So, artists.map {|a| a.name} will give you an array of artist names.
A cute shortcut for that is:  artists.map(&:name) and if you want to know why, you can look here:

http://stackoverflow.com/questions/9468564/what-does-post-all-mapid-mean

But basically it is ok just to remember that little shortcut.

Then you call join and your array of names (to join them into a string) and pass it a comma and space separator.

:)